Bug in constant_sensor.h

I declare the following constant string sensor:

`auto *mode = new StringConstantSensor("managed", 1000, "/config/mode");`

I get the following error:

line 26 in main.cpp is the line above declaring the constant sensor.

It looks like it has something to do with ArduinoJson 6.2

```

In file included from src/main.cpp:26:
.pio/libdeps/esp32dev/SensESP/src/sensesp/sensors/constant_sensor.h: In instantiation of 'bool sensesp::ConstantSensor<T>::set_configuration(const ArduinoJson::V6215PB2::JsonObject&) [with T = String]':
.pio/libdeps/esp32dev/SensESP/src/sensesp/sensors/constant_sensor.h:72:16:   required from here
.pio/libdeps/esp32dev/SensESP/src/sensesp/sensors/constant_sensor.h:75:14: error: ambiguous overload for 'operator=' (operand types are 'String' and 'ArduinoJson::V6215PB2::detail::enable_if<true, ArduinoJson::V6215PB2::detail::MemberProxy<ArduinoJson::V6215PB2::JsonObject, const char*> >::type' {aka 'ArduinoJson::V6215PB2::detail::MemberProxy<ArduinoJson::V6215PB2::JsonObject, const char*>'})
       value_ = config["value"];
In file included from .pio/libdeps/esp32dev/NMEA2000-library/src/N2kDef.h:57,
                 from .pio/libdeps/esp32dev/NMEA2000-library/src/N2kStream.h:39,
                 from .pio/libdeps/esp32dev/NMEA2000-library/src/N2kMsg.h:44,
                 from .pio/libdeps/esp32dev/NMEA2000-library/src/N2kMessages.h:57,
                 from src/main.cpp:13:
/Users/johan/.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:104:18: note: candidate: 'String& String::operator=(const String&)'
         String & operator =(const String &rhs);
                  ^~~~~~~~
/Users/johan/.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:105:18: note: candidate: 'String& String::operator=(const char*)'
         String & operator =(const char *cstr);
                  ^~~~~~~~
/Users/johan/.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:106:18: note: candidate: 'String& String::operator=(const __FlashStringHelper*)'
         String & operator = (const __FlashStringHelper *str) {return *this = reinterpret_cast<const char*>(str);}
                  ^~~~~~~~
/Users/johan/.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:108:18: note: candidate: 'String& String::operator=(String&&)'
         String & operator =(String &&rval);
                  ^~~~~~~~
/Users/johan/.platformio/packages/framework-arduinoespressif32/cores/esp32/WString.h:109:18: note: candidate: 'String& String::operator=(StringSumHelper&&)'
         String & operator =(StringSumHelper &&rval);
```

--- a/src/sensesp/sensors/constant_sensor.cpp
+++ b/src/sensesp/sensors/constant_sensor.cpp
@@ -1,0 +1,16 @@
+#include "constant_sensor.h"
+
+#include <Arduino.h>
+
+namespace sensesp {
+
+template <>
+const String ConstantSensor<String>::sensor_type_ = "string";
+template <>
+const String ConstantSensor<int>::sensor_type_ = "integer";
+template <>
+const String ConstantSensor<float>::sensor_type_ = "number";
+template <>
+const String ConstantSensor<bool>::sensor_type_ = "boolean";
+
+}  // namespace sensesp

--- a/src/sensesp/sensors/constant_sensor.h
+++ b/src/sensesp/sensors/constant_sensor.h
@@ -40,10 +40,10 @@ namespace sensesp {
  * @param[in] config_path Configuration path for the sensor.
  */
 
-static const char SCHEMA_CONSTANT_SENSOR[] PROGMEM = R"###({
+static const char SCHEMA_CONSTANT_SENSOR[] = R"###({
         "type": "object",
         "properties": {
-            "value": { "title": "Constant Value", "type": "number", "description": "Constant value" }
+            "value": { "title": "Constant Value", "type": "%TYPE%", "description": "Constant value" }
         }
     })###";
 
@@ -72,7 +72,7 @@ class ConstantSensor : public Sensor<T> {
   virtual bool set_configuration(const JsonObject &config) override {
     // Neither of the configuration parameters are mandatory
     if (config.containsKey("value")) {
-      value_ = config["value"];
+      value_ = config["value"].as<T>();
     }
     if (config.containsKey("interval")) {
       send_interval_ = config["interval"];
@@ -80,13 +80,28 @@ class ConstantSensor : public Sensor<T> {
     return true;
   }
   virtual String get_config_schema() override {
-    return FPSTR(SCHEMA_CONSTANT_SENSOR);
+    String schema = SCHEMA_CONSTANT_SENSOR;
+    schema.replace("%TYPE%", sensor_type_);
+    return schema;
   }
   void update() { this->emit(value_); }
 
   T value_;
   int send_interval_;  // seconds
+
+  static const String sensor_type_;
 };
+
+template <class T>
+const String ConstantSensor<T>::sensor_type_ = "";
+template <>
+const String ConstantSensor<int>::sensor_type_;
+template <>
+const String ConstantSensor<float>::sensor_type_;
+template <>
+const String ConstantSensor<String>::sensor_type_;
+template <>
+const String ConstantSensor<bool>::sensor_type_;
 
 // ..........................................
 //  constant value sensors

